tools/convert_checkpoint/weights2megatron/weights2megatron_llama.py

```diff
@@ -0,0 +1,186 @@
+import os
+import sys
+import shutil
+from pathlib import Path
+from typing import Optional
+from argparse import ArgumentParser, Namespace
+
+import torch
+from tqdm.auto import trange
+from transformers import AutoModelForCausalLM, LlamaTokenizer
+
+from permute_qkv import permute_qkv
+from merge_llama import merge_llama
+from transformers import AutoTokenizer
+
+llama_s2layer = {7: 32, 13: 40, 30: 60, 65: 80, 70: 80}
```

Comment:
I am concerned about the frequent use of hard-coded constants. Could this information not be inferred from the transformers checkpoint file? Alternatively, couldn't LLamaConfig be used rather than these constants?

Reply:
You are right, using LLamaConfig is a better solution. I will make the necessary modifications accordingly.